docs/product_release.md

```diff
@@ -0,0 +1,18 @@
+# Product Release
```

Comment:
Maybe it's worth mentioning that products have their release processes and to refer to the according product specific repositories for information about those.